BrainPortal/lib/cbrain_task_generators/templates/portal.rb.erb

```diff
@@ -261,7 +266,55 @@ class CbrainTask::<%= name %> < <%= descriptor['cbrain:inherits-from-class'] ||
     # Check that disables-inputs is not violated
     # If the parameter is active, the ones it disables must not be
 %    generateCheckerLoop.("disable")
-% end   
+% end
+%#
+%####
+%# Check for violations pertaining to parameter groups
+%# In particular, check that mutual exclusivity and one-is-required are satisfied 
+%#
+% unless noGroups
+%# Check if any groups require checking
+% getGroupsWith = lambda { |prop| groups.select{ |g| g[prop] }.map{ |g| ":'"+g["id"]+"'" }.join(", ") }
+% mutexGrps = getGroupsWith.("mutually-exclusive") 
+% oneReqGrps = getGroupsWith.("one-is-required")
+% hasNoMutex = (mutexGrps == "")
+% hasNoOneReq = (oneReqGrps == "")
+%#
+% unless (hasNoMutex && hasNoOneReq)
+   
+    # Groups with the mutually exclusive or one-is-required
+    <%= 'mutexGroups = [%s]'  % mutexGrps  unless hasNoMutex %>
+    <%= 'oneReqGroups = [%s]' % oneReqGrps unless hasNoOneReq %>
+    # Mapping from groupId to members list
+%   writeLiteralMap.("gidToMbrs", gIdToMbrs)
+    # Mapping from groupId to group name
+    grpName = {<%= groups.map{ |g| ":'" + g["id"] + "\' => \'" + g["name"] + "'" }.join(", ") %>}
+%  unless hasNoMutex
```

Comment:
style: what about 'if hasMutex'? hasNoMutex and hasNoOneReq are used only with unless, double negations could be avoided.